commercetools-sdk-base/src/test/java/io/sphere/sdk/queries/PagedQueryResultTest.java

```diff
@@ -44,6 +46,32 @@ public void middleOfManyResult() throws Exception {
     }
 
     @Test
+    public void getResultPage() throws Exception {
+        final PagedQueryResult<Integer> queryResult = PagedQueryResult.of((long) 80, TOTAL, listOfSize(PAGE_SIZE));
+        assertThat(queryResult.getPage()).isEqualTo(3);
```

Comment:
👍 very good catch actually ny algo was bugus i guess, 
for the remarque: 
>request past the last page (offset = 125, total = 106, count = 0, page = ??
i think the best solution is to throw an 'IllegalStateException' with message 'Page number greater then total number of pages' and then even if its a misbehaviour from the backend it should be raised to be corrected.
What do you think?